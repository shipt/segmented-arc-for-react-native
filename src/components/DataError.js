/* eslint-disable react-native/no-color-literals */
// Disabled because if a user wants to override it, they can pass dataErrorComponent with any customization
import PropTypes from 'prop-types';
import React from 'react';
import { Text, View, StyleSheet } from 'react-native';

const DataError = ({ testID, style }) => {
  return (
    <View style={[styles.container, style]} testID={testID}>
      <Text style={styles.text}>An issue occurred while calculating the graph data, so it may be inaccurate.</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#F3F4F6',
    borderRadius: 6,
    margin: 4,
    marginVertical: 8,
    padding: 4
  },
  text: {
    color: '#000000',
    textAlign: 'center'
  }
});

DataError.propTypes = {
  testID: PropTypes.string,
  style: PropTypes.object
};

export default DataError;
