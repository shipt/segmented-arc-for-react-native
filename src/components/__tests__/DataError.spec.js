import { render } from '@testing-library/react-native';
import React from 'react';
import { StyleSheet } from 'react-native';
import DataError from '../DataError';

describe('DataError', () => {
  it('renders default', () => {
    expect(render(<DataError />)).toMatchSnapshot();
  });

  it('renders with custom style', () => {
    const style = StyleSheet.create({ width: 200 });
    const wrapper = render(<DataError style={style} testID="data-error" />);
    expect(wrapper.getByTestId('data-error')).toHaveStyle(style);
  });
});
