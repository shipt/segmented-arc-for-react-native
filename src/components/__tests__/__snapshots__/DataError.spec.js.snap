// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataError renders default 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#F3F4F6",
        "borderRadius": 6,
        "margin": 4,
        "marginVertical": 8,
        "padding": 4,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "textAlign": "center",
      }
    }
  >
    An issue occurred while calculating the graph data, so it may be inaccurate.
  </Text>
</View>
`;
